interfaces/systemd/backend.go

```diff
@@ -69,27 +97,31 @@ func (b *Backend) Setup(snapInfo *snap.Info, devMode bool, repo *interfaces.Repo
 		return fmt.Errorf("cannot create directory for systemd services %q: %s", dir, err)
 	}
 	glob := interfaces.InterfaceServiceName(snapName, "*")
-	changed, removed, errEnsure := osutil.EnsureDirState(dir, glob, content)
+
 	systemd := sysd.New(dirs.GlobalRootDir, &dummyReporter{})
+	// We need to be carefully here and stop all removed service units before
+	// we remove their files as otherwise systemd is not able to disable/stop
+	// them anymore.
+	if err := stopAndDisableAllRemovedServices(systemd, glob, content); err != nil {
+		logger.Noticef("cannot stop removed services: %s", err)
+	}
+	changed, removed, errEnsure := osutil.EnsureDirState(dir, glob, content)
 	// Reload systemd whenever something is added or removed
 	if len(changed) > 0 || len(removed) > 0 {
 		err := systemd.DaemonReload()
 		if err != nil {
 			logger.Noticef("cannot reload systemd state: %s", err)
 		}
 	}
-	// Start any new services
+	// Start and enable or restart any new or changed services
```

Comment:
Let's please rephrase it so it's saying what we want instead of what the code written below reads as:
```
// Ensure the service is running right now and on reboots.